core/lib/zksync_core/src/api_server/tx_sender/mod.rs

```diff
@@ -629,6 +631,7 @@ impl TxSender {
         mut tx: Transaction,
         estimated_fee_scale_factor: f64,
         acceptable_overestimation: u32,
+        state_override: Option<StateOverride>,
```

Comment:
I'd prefer to have an enum here with 2 options (state/statediff) and check the correctness on api level. 
At least error doesn't make sense in this case, because it's not a tx failure, it's a request failure

Reply:
Changed in e2ad0f4c1690bbe45e1b43268a170c7878b99122